### ISSUE TITLE
Bug in active detection?

### ISSUE THREAD

**Issue by afmagee42** [CDCgov/cfa-ring-vax-widget#33]
Using #32, moving the active detection probability slider around has very little impact on the outcomes.

At 500 simulations, otherwise default settings, I'm seeing:
- Active prob 4%: control prob 68%, average duration of infectiousness 2.04, $R_e$ 0.98, prob of active detection 0
- Active prob 49%: control prob 68%, average duration of infectiousness 2.04, $R_e$ 0.98, prob of active detection 0.02
- Active prob 82%: control prob 70%, average duration of infectiousness 2.01, $R_e$ 0.96, prob of active detection 0.05

Why is active detection not happening?

**Comment by afmagee42** [comment]
[Some thinking aloud]

Since the passive detection delay is 2, we can potentially actively detect new infections that arise in the first unit of time of the duration of infectiousness of a passively-detected individual. This on average is 0.5 infections (birth rate being used is 0.5). Thus I would expect something like 0.25 active detections per passive detection, ignoring active-active chains of detection.

Since the active detection delay is 2, and the latent duration is 1, any such actively-detected infection should arise in its own first unit of infectious time. Since infection times are uniform, this is an average of 0.25 new infections which could be actively detected after an active detection.

**Comment by afmagee42** [comment]
Maybe this isn't wrong but just deeply counter-intuitive?

Fixing delays and rates, if I passively detect a case, on average
- It's caused 0.5 infections
- I detect `p_active_detect` \* 0.5 next-infections actively
- I detect `p_passive_detect` \* (1 - `p_active_detect`) \* 0.5 next-infections because active happens first when it happens


If I _don't_ passively detect a case
- It causes on average 1.5 next-infections 
- I actively detect 0 of these, because I missed this infection
- I passively detect `p_passive_detect` \* 1.5

So, the marginal proportion of next-infections passively detected isn't hugely affected by whether or not a particular infection is detected (we lose a small chunk of a small number to active detection). But the reverse is not true, because for every infection we miss, we get a lot more infections we could never actively detect.

**Comment by afmagee42** [comment]
Paging @swo because my brain now hurts

**Comment by swo** [comment]
I think my first antidote is #14 ; we should have some unit tests for the "do detections" function, and make sure it's doing what we think it should

I think my second antidote is #23 (unless overcome by another issue), so we can visualize what is happening in the simulations!

**Comment by afmagee42** [comment]
Agree that those will help make sure that the "problem" is actually just unintuitive emergent behavior.

**Comment by swo** [comment]
> we can get a passive-passive chain when active detection probability is 100% and active detection delay is 0.0.

I think this is OK? (Unless you meant that *passive* detection delay were 0?) Two people in a chain can be passively detected after they transmit.

**Comment by afmagee42** [comment]
Right. The later infection is actually detected first. Generation-views hurt my brain.

**Comment by afmagee42** [comment]
I bet this is related to #53. That bug would (sort of) make active vaccination behave as if it had a random delay from t_exposed that was 1 + Uniform(0, 3), so it would lose out to passive when it shouldn’t.

**Comment by swo** [comment]
Agree that #53 likely will solve this

And/or we can use #55 to see what is going on!

**Comment by afmagee42** [comment]
I'm thinking we probably also just want some tests? It'll be a stab in the dark as any test is based on something stochastic, but better than nothing perhaps.
- Set all detection probabilities to 1, active delay to 0, passive delay to (exposed duration + infectious duration - epsilon). Every case except the index case should be actively detected.
- Now set active delay to be $\approx \infty$. No active detections should happen, but all cases should be passively detected.

**Comment by afmagee42** [comment]
We should also probably consider some less unit-testing, more scientific correctness testing when we think the code base is stable. Run many (like 10k) simulations where. Do the math by hand to figure out the actual active detection probability (Pr(active and active happens first), depends on active and passive delays and a uniform distribution over infection times) and check that the active detection probability is close-ish to what it should be.

**Comment by afmagee42** [comment]
I think #73 has me sufficiently convinced that the problem was #53 that we can close this when #55 is merged. (Tests still wouldn't hurt, but we can separate that out in a new issue.)
